everything-cat/src/api/index.js

```diff
@@ -0,0 +1,95 @@
+import axios from 'axios';
```

Comment:
Why did you decide to use axios over the native fetch API?

Reply:
I chose axios for its built-in support for timeouts, interceptors, and error handling. It also simplifies tasks like setting headers and parsing JSON. While these features may not be essential for a simple app, I selected it with scalability in mind for future growth.

Reply:
Fair enough. But are any of these features necessary in your app?

Reply:
In the current project it is true that is not needed and I initially used only fetch. I made the switch to axios because:

- I called `POST:https://api.thecatapi.com/v1/favourites` with an existing image id and it returned 400. To me that was an indication that the handling error functionality from AXIOS would come in handy in order to implement both a generic errors handler and specific requests like treating a 400 as non-fatal
- Another reason for switching was the POST to https://api.thecatapi.com/v1/images/search, which returned more data than needed. Using Axios’s `transformResponse `would allow to filter the response before passing it to the component.

Also i since it is fairly lightweight in bundle size (35.8 kB Minified) I believed that the future benefits would overscale the extra bundle size.

To be fair, Axios isn't really needed for this project as said earlier, and if I followed the YAGNI approach, I wouldn't have used it. However, I chose it with future growth in mind, anticipating the app will scale and benefit from its features later on.